### PR TITLE
Develop IP offset for hosts

### PR DESCRIPTION
**Pull request by anastop**

### CODE REVIEW

File: multi/master.py [hunk 1]
```diff
@@ -110,9 +111,12 @@ def get_flows():
         requests.models.Response: An HTTP Response with the aggregated
         status codes and bodies of the broadcasted requests
     """
+    t_start = time.time()
     reqs = m_util.broadcast_cmd(WORKER_IP_LIST, WORKER_PORT_LIST,
                                 'get_flows')
     stat, bod = m_util.aggregate_broadcast_response(reqs)
+    flow_latency_interval_on_master = time.time() - t_start
```
Comment (anastop) [inline multi/master.py]: rename this variable to `get_flows_latency`.

File: multi/master.py [hunk 2]
```diff
@@ -110,9 +111,12 @@ def get_flows():
...
+    logging.info('flow_latency_interval_on_master:{0} [sec]'.format(flow_latency_interval_on_master))
```
Comment (anastop) [inline multi/master.py]: rephrase the log msg: `'get_flows_latency on master...'`

File: net/multinet.py [hunk 3]
```diff
@@ -281,12 +298,17 @@ def get_flows(self):
...
+        t_start = time.time()
+
         for switch in self.switches:
             flows_list = switch.dpctl('-O OpenFlow13 dump-flows').split('\n')
             flow_number = len(flows_list) - 2
             flow_number_total += flow_number
 
         logging.debug('[get_flows] number of flows: {0}'.format(flow_number_total))
+
+        flow_latency_interval_on_single_worker = time.time() - t_start
```
Comment (anastop) [inline net/multinet.py]: rename this to `get_flows_latency`

File: net/multinet.py [hunk 4]
```diff
@@ -281,12 +298,17 @@ def get_flows(self):
...
+        logging.info('flow_latency_interval_on_single_worker:{0} [sec]'.format(flow_latency_interval_on_single_worker))
```
Comment (anastop) [inline net/multinet.py]: also this

File: net/multinet.py [hunk 5]
```diff
@@ -334,13 +355,14 @@ def generate_traffic(self):
...
         while (time.time() - transmission_start) <= traffic_transmission_interval:
             src_mac, dst_mac = self.generate_mac_address_pairs(current_mac)
-
             current_mac = hex(int(current_mac, 16) + 2)
-            self.hosts[host_index].sendCmd('sudo mz -a {0} -b {1} -t arp'.
-                                           format(src_mac, dst_mac))
-            self.hosts[host_index + 1].sendCmd('sudo mz -a {0} -b {1} -t arp'.
-                                               format(dst_mac, src_mac))
-            time.sleep(traffic_transmission_delay)
+
```
Comment (anastop) [inline net/multinet.py]: Can you provide some __short__ comments on the way that traffic generation works, and other things like e.g. splitting the generation in two phases?

File: net/multinet.py [hunk 6]
```diff
@@ -334,13 +361,31 @@ def generate_traffic(self):
...
-                                           format(src_mac, dst_mac))
-            self.hosts[host_index + 1].sendCmd('sudo mz -a {0} -b {1} -t arp'.
-                                               format(dst_mac, src_mac))
-            time.sleep(traffic_transmission_delay)
+            # This is the place where generation of flows happens. In order to
+            # work properly we must configure the ODL controller with L2Switch
+            # plugin and each switch of the topology must have at least 2
+            # hosts.
+            # Step1:
+            # From host1 of switch1 we initially send a Gratuitous ARP Reply.
```
Comment (anastop) [inline net/multinet.py]: is this a Reply or Request?

File: net/multinet.py [hunk 7]
```diff
@@ -334,13 +361,31 @@ def generate_traffic(self):
...
+            # generator in this class.
+            # Step2:
+            # We repeat the above steps from host2 of switch1 reversing the src
+            # and dst MAC addresses of the ethernet frame.
+            # The above sequence has as a result to trigger ODL controller to
+            # respond with 2 FlowMod messages in order to establish a datapath
+            # between the 2 hosts
+            self.hosts[host_index].sendCmd(
+                'sudo mz -a {0} -b {1} -t arp'.format(src_mac, dst_mac))
+            # We brake transmission delay and we place a delay between the
```
Comment (anastop) [inline net/multinet.py]: ~~brake~~ --> break

File: util/multinet_requests.py [hunk 8]
```diff
@@ -155,7 +155,7 @@ def broadcast_cmd(worker_ip_list, worker_port_list, opcode, data=None):
                                                 result_queue,))
         processes.append(process)
         process.start()
-        time.sleep(1)
+        #time.sleep(1)
```
Comment (anastop) [inline util/multinet_requests.py]: delete this entirely

Comment (konstantinos-papadopoulos) [inline util/multinet_requests.py]: deleted

File: net/multinet.py (hunk 6 above)
Comment (konstantinos-papadopoulos) [inline net/multinet.py]: It is **Reply**
http://www.perihel.at/sec/mz/mzguide.html#arp

By default an ARP reply is sent with your own interface addresses as source MAC and IP address, and a broadcast destination MAC/IP address.

File: util/multinet_requests.py (hunk 8 above)
Comment (konstantinos-papadopoulos) [inline util/multinet_requests.py]: Deleted

**Pull request merged**